src/replication.c

```diff
@@ -1970,7 +1970,7 @@ void replicationCron(void) {
     }
 
     /* Timed out master when we are an already connected slave? */
-    if (server.masterhost && server.repl_state == REDIS_REPL_CONNECTED &&
+	if (server.masterhost && server.repl_state == REDIS_REPL_CONNECTED && server.master &&
```

Comment:
This looks like an issue that's not specific to Windows. Could you elaborate on what the bug is? Would it make sense to submit this to antirez\redis?